test/unit/manipulation.js

```diff
@@ -500,6 +500,53 @@ QUnit.test( "html(String) tag-hyphenated elements (Bug #1987)", function( assert
 	assert.equal( j.children().text(), "text", "Tags with multiple hypens behave normally" );
 } );
 
+QUnit.test( "html(String) respect the spec on how to handle special and name-terminating chars for tag names (gh-2005)", function( assert ) {
+
+	assert.expect( 120 );
+
+	var wrapper, child, nodeName,
+		nameTerminatingChars = "\x20-\t-\r-\n-\f".split( "-" ),
+		specialChars = "[ ] { } _ - = + \\ ( ) * & ^ % $ # @ ! ~ ` ' ; ?".split( " " );
```

Comment:
Surely - added 10 non-ASCII characters to the list. Please let me know if adding more is needed.